src/reader.rs

```diff
@@ -6511,10 +6511,22 @@ impl<'a> Reader<'a> {
             prefix.push_utfstr(&tok);
             prefix.push_utfstr(&common_prefix);
         } else {
-            // Append just the end of the string.
+            // Collapse parent directories and append end of string
```

Comment:
> This didn't break any tests locally for me. I'm willing to add new tests for this behavior, but I could use some direction as to where they would need to go and what sort of setup and/or mocking would be required.

We don't have mocking for `src/reader.rs` yet, so either add that or write
a system test, maybe in `tmux-pager.fish`. To run, use

	cargo b && tests/test_driver.py target/debug tests/checks/tmux-pager.fish